docs/dev/maintainer.md

```diff
@@ -179,7 +179,8 @@ latest submodules.
     updated.
  3. Edit the submodule update commits with any necessary additional
     information. For example, amend the message with `Fixes` tags.
```

Comment:
> Fixes: https://github.com/scylladb/scylladb/commit/71da4e6e795f2204bde76473d4876479c2e35692

While I deeply wish that all commits would indicate which earlier commit they correct (as a more precise and visible alternative to backport/branch labels), the Fixes keyword is now used to indicate issues, not commits.

We need to find a distinct keyword here. Maybe Originally-Broken-By:? Too verbose.

Reply:
You don't really need a "keyword", you can just use natural English text as you see fit. I often write in a commit message text like "In commit  12345678 we did X and Y, but it's wrong so we fix it here". Since we don't have bots that need to understand that one commit fixed another, I don't think we need a new keyword.

Reply:
I want a keyword so automation can make use of it.

Reply:
How about `Updates: ` or `Corrects: ` (or even `Repairs: `)?

Reply:
Avi, your suggestion has two problems:
1. We *don't* have automation of the sort you are suggesting that uses this "commit x is related to commit y", and it's not clear to me (maybe it is clear to you?) why we'd even want such automation - what would it do?
2. Even if we did have such automation, because this "commit x is related to commit y" is so rare, nobody will remember its keyword. So if you call it "Corrects:", who will remember it? It's not like this "Fixes" keyword which is used in most of the commits.

Finally, I want to comment that github's automation doesn't need these "keywords" to detect links - it detects a any string of bytes that looks like a commit hash, and if it matches an existing commit, it "makes the link". Our hypothetical automation could do the same.

Reply:
I think such a `Fixes: commit-hash` syntax is useful when backporting. If you backport commit C1 from master to a stable branch, you can at once search the rest of the master branch for `Fixes: C1`. If you find any such commit, you can backport those at once as well. Otherwise, backporting only C1 might fix one issue on the stable branch, but introduce another (i.e., a regression). It's a somewhat greppable syntax for attaching reminders to existent commits. (And then the next question would be the minimum number of nibbles to use in such a commit hash reference. 7? 8? 12? 40? I tend to use 12 everywhere.)

Reply:
@lersek while in theory you are correct, in practice, at least in my experience (please correct me if you think my memory is playing tricks of me), only a tiny minority of commits mention other commits, so this trick will have a negligible effect on our backporting experience, even if people do remember to use the new keyword you're suggesting (and I'm claiming that they won't remember).

In any case, as I said earlier, the **keyword** isn't necessary: commit hashes look like random strings so if you want you can already "grep" them in git log, you don't need a keyword like "Fixes: " in front of them to make them less ambiguous.

Reply:
@nyh

I agree that the usefulness of `Fixes: commit` does depend on its disciplined use. I strive to use it when a clear (or at least very likely) culprit is known, usually from git-blame (or even git-bisect), and I tend to look for "culprits" in general -- I feel that digging into the history of the project simply lets me write better patches, or at least better-reasoned commit messages. If you predict that most of us will forget this "custom", then I agree it's mostly useless. (At the same time, it's also *harmless* if someone adds entries like this, as long as their "keyword of choice" does not conflict with another established keyword.)

... I tend to disagree with your point about disambiguation. Once I get to know a project a bit better, I tend to reference a bunch of commits in my commit messages, simply as a way of educating myself, and for explaining my train of thought to my reviewers, so they can catch my mistakes more easily. Those commit references are definitely not assigning blame, instead, they mark "sources of truth". Therefore I feel there's value in formally (greppably) distinguishing a reference that attributes breakage (or a missed cleanup opportunity) to a commit.

Anyway, I'm happy to go with the flow; I can drop the `Fixes` line, if you and Avi agree on that. Thanks.

Reply:
I've dropped the `Fixes: ` line from the commit message of v2, for now.